plugins/design-system/components/component-example/ComponentExample.js

```diff
@@ -0,0 +1,244 @@
+import CodeExampleTabs from "./CodeExampleTabs";
+import CodeExampleTabView from "./CodeExampleTabView";
+import CodeExampleTabViewList from "./CodeExampleTabViewList";
+
+import CodeExampleTabButton from "./CodeExampleTabButton";
+
+import { ReactCode, HtmlCode, Preview } from "./ComponentExampleOverrides";
+
+const DEFAULT_HEIGHT = 205;
+
+const REACT_STRING_OPTIONS = {
+  showDefaultProps: false,
+  useBooleanShorthandSyntax: false
+};
+
+const TAB_LABELS = {
+  React: "REACT",
+  Html: "HTML"
+};
+
+const METHODS_TO_BIND = [
+  "expandCollapseToggle",
+  "handleTabChange",
+  "handleCodeCopy",
+  "handleCanExpand"
+];
+
+class ComponentExample extends Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      isExpanded: false,
+      isCodeCopied: false,
+      canExpand: false,
+      activeTab: this.isTabbedMode() ? this.props.activeTab || ReactCode : null
+    };
+
+    this.defaultHeight = this.props.defaultHeight || DEFAULT_HEIGHT;
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  findChildByType(type) {
+    return [].concat(this.props.children).find(function(child) {
+      return child.type === type;
+    });
+  }
+
+  isEmptyHtmlOverride() {
+    const htmlCode = this.findChildByType(HtmlCode);
+
+    return htmlCode && !htmlCode.props.children;
+  }
+
+  isEmptyReactOverride() {
+    const reactCode = this.findChildByType(ReactCode);
+
+    return reactCode && !reactCode.props.children;
+  }
+
+  isPreviewMode() {
+    return this.isEmptyHtmlOverride() && this.isEmptyReactOverride();
+  }
+
+  isSingleBlockMode() {
+    return (
+      !this.isPreviewMode() &&
+      (this.isEmptyHtmlOverride() || this.isEmptyReactOverride())
+    );
+  }
+
+  isReactMode() {
+    return this.isSingleBlockMode() && this.isEmptyHtmlOverride();
+  }
+
+  isShowingReactCode() {
+    return this.isReactMode() || this.state.activeTab === ReactCode;
+  }
+
+  isTabbedMode() {
+    return !this.isSingleBlockMode();
+  }
+
+  getHtmlCode(jsxCode) {
+    let htmlCode = this.findChildByType(HtmlCode);
+    if (htmlCode) {
+      htmlCode = htmlCode.props.children;
+    }
+
+    return StringUtil.formatMarkdown(
+      ReactDOMServer.renderToStaticMarkup(htmlCode || jsxCode)
+    );
+  }
+
+  getReactCode(jsxCode) {
+    let reactCode = this.findChildByType(ReactCode);
+    if (reactCode) {
+      reactCode = reactCode.props.children;
+    }
+
+    return reactElementToJSXString(reactCode || jsxCode, REACT_STRING_OPTIONS);
+  }
+
+  expandCollapseToggle() {
+    this.setState({ isExpanded: !this.state.isExpanded });
+  }
+
+  handleTabChange(activeTab) {
+    this.setState({
+      activeTab,
+      isExpanded: false
+    });
+  }
+
+  handleCanExpand(canExpand) {
+    this.setState({ canExpand });
+  }
+
+  handleCodeCopy() {
+    this.setState({ isCodeCopied: true });
+  }
+
+  generateCodeExample(lang, code) {
```

Comment:
I'd replaced this function with a `getCodeBody` function and moved everything there.